test/jdk/java/awt/Window/OwnedWindowsLeak/OwnedWindowsLeak.java

```diff
@@ -70,31 +71,22 @@ public static void main(String[] args)
         // Third, make sure all the weak references are null
         for (WeakReference<Window> ref : children)
         {
-            if (ref.get() != null)
-            {
-                throw new RuntimeException("Test FAILED: some of child windows are not GCed");
+            while (ref.get() != null) {
+                System.out.println("ref.get() = " + ref.get());
+                System.gc();
+                Thread.sleep(1000);
             }
         }
 
         // Fourth, make sure owner's children list contains no elements
-        try
-        {
-            Field f = Window.class.getDeclaredField("ownedWindowList");
-            f.setAccessible(true);
-            Vector ownersChildren = (Vector)f.get(owner);
-            if (ownersChildren.size() > 0)
-            {
-                throw new RuntimeException("Test FAILED: some of the child windows are not removed from owner's children list");
-            }
-        }
-        catch (NoSuchFieldException z)
-        {
-            System.out.println("Test PASSED: no 'ownedWindowList' field in Window class");
-            return;
-        }
-        catch (Exception z)
+        Field f = Window.class.getDeclaredField("ownedWindowList");
+        f.setAccessible(true);
+        Vector ownersChildren = (Vector)f.get(owner);
+        while (ownersChildren.size() > 0)
         {
-            throw new RuntimeException("Test FAILED: unexpected exception", z);
+            System.out.println("ownersChildren = " + ownersChildren);
```

Comment:
This private list is cleared via the DIsposer mechanism, which may require additional time. If it fails then it will be 100% product bug.